### PR TITLE
Try adding a stack buffer based bulk method to IteratorProtocol

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/ContiguousArray.swift [hunk 1]
```diff
@@ -832,6 +832,7 @@ extension ContiguousArray: RangeReplaceableCollection {
   ///   `newElements`, over many calls to `append(contentsOf:)` on the same
   ///   array.
   @inlinable
+  @inline(never)
```
Comment (Catfish-Man) [inline stdlib/public/core/ContiguousArray.swift]: …dammit, that will totally invalidate benchmark results

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Pull request closed without merging**